Fails to compile in Angular 12

### What happened?

After installing Markdoc in a new angular 12 app, I get the following error.

```
Error: node_modules/@markdoc/markdoc/dist/index.d.ts:104:48 - error TS2694: Namespace '"/Users/jandrews/Projects/md/node_modules/@markdoc/markdoc/dist/src/types"' has no exported member 'Function'.

104         Function: typeof import("./src/types").Function;
```

### To reproduce

1) Install Angular 12.
2) Create new project `ng new markdoc-test`
3) Install markdoc `npm install @markdoc/markdoc`
4) Install dependencies `npm i --save-dev @types/react` and `npm i --save-dev @types/markdown-it`
5) Run Angular `ng serve --ssl`

The server will start and fail with:

```
Error: node_modules/@markdoc/markdoc/dist/index.d.ts:104:48 - error TS2694: Namespace '"/Users/jandrews/Projects/md/node_modules/@markdoc/markdoc/dist/src/types"' has no exported member 'Function'.

104         Function: typeof import("./src/types").Function;
```

The problem seems to be `Function` isn't exported, and since it's imported in, it can't be exported with export statement.  I had edited `types.d.ts` to include `export class astFunction extends Function {}` and then the `index.ts` file on line 104 to use `        Function: typeof import("./src/types").astFunction;` instead, and that prevented the compile error.  Not sure if there is a better fix, but I thought I'd mention this here. 



### Version

^0.1.1

### Additional context

_No response_

--- a/src/types.ts
+++ b/src/types.ts
@@ -1,10 +1,9 @@
+import type Func from './ast/function';
 import type Node from './ast/node';
-import type Function from './ast/function';
 import type Tag from './ast/tag';
 
-export type { default as Function } from './ast/function';
-export type { default as Node } from './ast/node';
-export type { default as Tag } from './ast/tag';
+export type { Node, Tag };
+export declare type Function = Func;
 
 export interface AstType {
   readonly $$mdtype: 'Function' | 'Node' | 'Variable';
@@ -34,7 +33,7 @@ export type ConfigFunction = {
   returns?: ValidationType | ValidationType[];
   parameters?: Record<string, SchemaAttribute>;
   transform?(parameters: Record<string, any>, config: Config): any;
-  validate?(fn: Function, config: Config): ValidationError[];
+  validate?(fn: Func, config: Config): ValidationError[];
 };
 
 export interface CustomAttributeTypeInterface {

